src/nidigital/system_tests/test_system_nidigital.py

```diff
@@ -12,463 +12,427 @@
-    assert all(len(fetched_waveforms[site]) == num_samples for site in fetched_waveforms)
+        multi_instrument_session.write_source_waveform_site_unique(
+            waveform_name='src_wfm',
+            waveform_data=source_waveform)
 
+        multi_instrument_session.pins['HighPins'].create_capture_waveform_parallel(waveform_name='capt_wfm')
 
-@pytest.fixture(params=[tuple, int, str])
-def source_waveform_wrong_type(request):
-    return request.param
+        multi_instrument_session.burst_pattern(start_label='new_pattern')
 
+        # Pattern burst is configured to fetch num_samples samples
+        fetched_waveforms = multi_instrument_session.fetch_capture_waveform(
+            waveform_name='capt_wfm',
+            samples_to_read=num_samples)
 
-def test_source_waveform_parallel_site_unique_wrong_type(multi_instrument_session, source_waveform_wrong_type):
-    '''Test methods for passing wrong types write_source_waveform_site_unique .
+        assert sorted(fetched_waveforms.keys()) == sorted([0, 1])
+        assert all(len(fetched_waveforms[site]) == num_samples for site in fetched_waveforms)
 
-    - create_source_waveform_parallel
-    - write_source_waveform_site_unique
-    '''
-    test_name = test_source_waveform_parallel_site_unique.__name__
-    configure_session(multi_instrument_session, test_name)
+    @pytest.fixture(params=[tuple, int, str])
+    def source_waveform_wrong_type(self, request):
+        return request.param
 
-    multi_instrument_session.load_pattern(get_test_file_path(test_name, 'pattern.digipat'))
+    def test_source_waveform_parallel_site_unique_wrong_type(self, multi_instrument_session, source_waveform_wrong_type):
+        '''Test methods for passing wrong types write_source_waveform_site_unique .
 
-    num_samples = 256
-    multi_instrument_session.write_sequencer_register(reg=nidigital.SequencerRegister.REGISTER0, value=num_samples)
+        - create_source_waveform_parallel
+        - write_source_waveform_site_unique
+        '''
+        test_name = self.test_source_waveform_parallel_site_unique.__name__
+        self.configure_session(multi_instrument_session, test_name)
 
-    multi_instrument_session.pins['LowPins'].create_source_waveform_parallel(
-        waveform_name='src_wfm',
-        data_mapping=nidigital.SourceDataMapping.SITE_UNIQUE)
+        multi_instrument_session.load_pattern(self.get_test_file_path(test_name, 'pattern.digipat'))
 
-    if source_waveform_wrong_type == tuple:
-        source_waveform = ([i for i in range(num_samples)], [i for i in reversed(range(num_samples))])
-    elif source_waveform_wrong_type == int:
-        source_waveform = num_samples
-    elif source_waveform_wrong_type == str:
-        source_waveform = {
-            str(1): [str(i) for i in range(num_samples)],
-            str(0): [str(i) for i in reversed(range(num_samples))]}
-    else:
-        assert False, "Invalid source waveform data type: {}".format(source_waveform_type)
```

Comment:
Wow, how was this even passing, before?

Reply:
The test itself passed fine because this is only in the string of an `assert False`, so it was never actually evaluated. But it was able to pass by flake8 because there's a `def source_waveform_type` in this file. (But now that that `def` is inside the class, you'd have to update it to `self.source_waveform_type` for it to get past flake8.)